app/controllers/reports/patient_lists_controller.rb

```diff
@@ -0,0 +1,44 @@
+class Reports::PatientListsController < AdminController
+  skip_after_action :verify_authorized, if: -> { current_admin.permissions_v2_enabled? }
+  after_action :verify_authorization_attempted, if: -> { current_admin.permissions_v2_enabled? }
+
+  def show
+    scope = if current_admin.permissions_v2_enabled?
+      if region_class == "facility_group"
+        authorize_v2 { current_admin.accessible_facility_groups(:view_pii) }
+      else
+        authorize_v2 { current_admin.accessible_facilities(:view_pii) }
```

Comment:
This authorize_v2 block feels unnecessary w/ the new permissions system...since the scope already handles the authorization for us.

Reply:
To clarify here: Not trying to block any progress 😁 . We can circle back around later after permissions v2 is shipped to improve things - just trying to clarify now that I'm getting some real experience w/ the new perms system.

Reply:
I think that's fine, you can totally go ahead and not use the `authorize_v2` if you think you don't need it. And since you don't have an `index` method, you can _probably_ avoid it. I'm not at all married to having it around, it's the least interesting thing in the entirety of the permissions work :)

But here are some concrete reasons why it could be helpful to have *something* like it:

1. When you are dealing with collections, you will never call `find_by`. This is true for all index actions in all controllers. This means you can potentially scope things that you don't have authority to scope.
2. Wrapping it in a block allows us a generic API to provide a guard-rail against actions that have not been authorized, it'll immediately throw an exception if you don't wrap your actions in a block. This is obviously not as powerful as Pundit, because the Pundit guardrails would explicitly work for the specific things you are scoping not an arbitrary call to a block.
3. It's easier to see that authorization is taken care of when reading code. It's explicit and consistent instead of a hidden catch-all on `RecordNotFound` or in every controller.

It adds *some* value, but maybe not a whole lot. It was definitely easier to migrate from old to new using this because it keeps the old functionality in parity.

This idea, and pundit, and the million other auth libraries in Rails are all generally based off of the [PERM metamodel](https://casbin.org/docs/en/how-it-works) from https://casbin.org.